Scripts/package.py

```diff
@@ -131,19 +131,19 @@ def build(self):
             "-f", "net5.0",
             "-o", self.buildDirectory,
             "-r", self.target,
-            "-c", "Checked"])
+            "-c", "Release"])
```

Comment:
Wouldn't the safer change be to make this "Debug" instead? Are we confident we're not changing behavior by changing to "Release" here?

Reply:
> Wouldn't the safer change be to make this "Debug" instead?

True

> Are we confident we're not changing behavior by changing to "Release" here?

The behavior change I expect from this is that `Contract.Assert` no longer has an effect. This could cause code to verify or compile instead of Dafny crashing, although I'm guessing 99% of the possible cases would still cause Dafny to crash but slightly later down the road.

I don't see a good path forward that starts with releasing a Debug build here. The end situation, as far as C# goes, should be to make a Release build and to get verification and compilation safety from the test suite. What steps can we take between that end state and now? We can turn on nullable types to replace the null checking contracts, but those contracts are among the ones that will cause Dafny to still crash but slightly later if you turn them off.

I think we should bite the bullet and build a Release build now, with the alternative being first switching to nullable types. If you prefer the latter let's do that.

Reply:
I agree that it would be better to release a Release build. Having a non-standard Checked configuration was definitely confusing and I'd rather align with best practices.

My only concern is that the CI is currently running the lit-based test suite on the same artifacts we release, i.e. by running `package.py` and using the result. That's a great thing to make sure we thoroughly test what we release, but it also means if we switch to a Release build the CI won't exercise those `Contract.Assert` statements except in unit tests, where the coverage is currently a lot lower.

Reply:
Ah but this is already true since Checked wasn't exercising the `Contract.Assert` statements either, so changing to Release doesn't change that either way. My concern is moot since it already happened a year ago. We'll revisit when we turn on the nullability support as mentioned in #1390.